install.sh

```diff
@@ -165,7 +196,30 @@ else
 
 fi
 
-echo -e "${COLOR_WARNING}Please restart your device to see effects...${NOCOLOR}"
-echo "Have fun, until next time, bye!"
+# Apply changes if the xochitl service is active
+if systemctl --quiet is-active xochitl.service 2> /dev/null; then
+    touch /tmp/rm-hacks-skip-onfailure
+    if ! try_restart_xochitl; then
+        # TODO - Move uninstall to be something run by remarkable-fail.service if /tmp/rm-hacks-skip-onfailure exists
+        #      - Have /tmp/rm-hacks-skip-onfailure be removed by xochitl upon successful startup
+        #      - Don't remove /tmp/rm-hacks-skip-onfailure in this script
+        #      - Rework try-restart-xochitl to wait for file to be removed or xochitl.service to fail
+        echo -e "${COLOR_ERROR}Failed to restart the user interface, uninstalling...${NOCOLOR}"
+        uninstall
+        if ! try_restart_xochitl; then
```

Comment:
The problem here is that you've uninstalled the override to remarkable-fail.service. So if there is an issue with xochitl service that persists, it will force a restart which still softbricks the device.

Reply:
> The problem here is that you've uninstalled the override to remarkable-fail.service. So if there is an issue with xochitl service that persists, it will force a restart which still softbricks the device.

At this point, something has gone very wrong anyway, as it shouldn't fail. I'm not really comfortable leaving the override in place either, as that's not a true uninstall. I think it might be worth just leaving it as-is, as this shouldn't ever happen if it was working previously, and I only added it just in case something extremely unlikely to happen happens.

Reply:
I agree with that. Especially since we're not mucking about with the binary itself. Something would have to go horribly wrong to get softbrick.